lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -54,6 +54,9 @@ namespace {
   static cl::opt<bool> UseAlive("souper-use-alive",
     cl::desc("Use Alive2 as the backend"),
     cl::init(false));
+  static cl::opt<bool> LSBPruning("souper-lsb-pruning",
+    cl::desc("Try to prune guesses by looking for a difference in LSB"),
+    cl::init(true));
```

Comment:
I'd say this is off by default until we've done some more extensive testing for effect on performance